rafx/examples/renderer_triangle/renderer_triangle.rs

```diff
@@ -0,0 +1,375 @@
+        .init();
+
+    run().unwrap();
+}
+
+fn run() -> RafxResult<()> {
+    //
+    // Init SDL2 (winit and anything that uses raw-window-handle works too!)
+    //
+    let sdl2_systems = sdl2_init();
+
+    //
+    // Create the api. GPU programming is fundamentally unsafe, so all rafx APIs should be
+    // considered unsafe. However, rafx APIs are only gated by unsafe if they can cause undefined
+    // behavior on the CPU for reasons other than interacting with the GPU.
+    //
+    let api = unsafe { RafxApi::new(&sdl2_systems.window, &Default::default())? };
+    let mut resources = Resources::default();
+    resources.insert(TimeState::new());
+
+    // Wrap all of this so that it gets dropped before we drop the API object. This ensures a nice
+    // clean shutdown.
+    {
+        //
+        // For this example, we'll run the `distill` daemon in-process. This is the most convenient
+        // method during development. (You could also build a packfile ahead of time and run from that)
+        //
+        let db_dir = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("examples/renderer_triangle/.assets_db");
+        let asset_dir = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("examples/renderer_triangle/assets");
+        let connect_string = "127.0.0.1:9999";
+
+        let asset_source = AssetSource::Daemon {
+            external_daemon: false,
+            daemon_args: AssetDaemonOpt {
+                db_dir,
+                address: connect_string.parse().unwrap(),
+                asset_dirs: [asset_dir].to_vec(),
+            },
+        };
+
+        let sdl2_window = &sdl2_systems.window;
+
+        resources.insert(VisibilityRegion::new());
+        resources.insert(ViewportsResource::default());
+
+        let demo_render_feature_plugin = Arc::new(DemoRenderFeaturePlugin::default());
+        demo_render_feature_plugin.legion_init(&mut resources);
+
+        let mut renderer_builder = RendererBuilder::default();
+        renderer_builder = renderer_builder
+            .add_asset(Arc::new(DemoRendererPlugin))
+            .add_render_feature(demo_render_feature_plugin);
+
+        let mut renderer_builder_result = {
+            let extract_resources = ExtractResources::default();
+
+            let render_graph_generator = Box::new(DemoRenderGraphGenerator);
+
+            renderer_builder.build(
+                extract_resources,
+                &api,
+                asset_source,
+                render_graph_generator,
+                None,
+            )
+        }?;
+
+        let (width, height) = sdl2_window.vulkan_drawable_size();
+        let swapchain_helper = SwapchainHandler::create_swapchain(
+            &mut renderer_builder_result.asset_manager,
+            &mut renderer_builder_result.renderer,
+            sdl2_window,
+            width,
+            height,
+        )?;
+
+        resources.insert(api.device_context());
```

Comment:
We should add lots of comments to this file to explain what all these steps are and why they're necessary/helpful

Reply:
Is that a TODO for now, or a TODO for later? :philosorator:

Reply:
`TODO` for later -- tracking with https://github.com/aclysma/rafx/issues/137